From 3eefb8d9b29a708fa38552e580547c5defdb42ff Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Tue, 28 Jan 2014 15:10:48 -0500
Subject: [x86] Makefile: Add stack frame check

Message-id: <1390921848-31000-1-git-send-email-prarit@redhat.com>
Patchwork-id: 75449
O-Subject: [RHEL7.0 PATCH BZ 1058753] x86, Makefile, Add stack frame check
Bugzilla: 1058753
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1058753
Brew:  http://brewweb.devel.redhat.com/brew/taskinfo?taskID=6960614

Add -Wframe-larger-than to arch/x86/Makefile.  Since we're going to be
doing a lot of RHEL work we might as well catch these errors when they
happen.

Seems lightweight:

Before patch:

real    2m51.460s
user    58m27.723s
sys     8m58.233s

After patch:

real    2m51.207s
user    58m28.006s
sys     8m58.257s

so there is no impact to the build time AFAICT.

Successfully tested by me.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/Makefile b/arch/x86/Makefile
index cf3f2aa..fdf3dd3 100644
--- a/arch/x86/Makefile
+++ b/arch/x86/Makefile
@@ -118,6 +118,9 @@ ifeq ($(CONFIG_KMEMCHECK),y)
  KBUILD_CFLAGS += $(call cc-option,-fno-builtin-memcpy)
 endif
 
+# RHEL frame check
+KBUILD_CFLAGS += -Wframe-larger-than=2048
+
 # Stackpointer is addressed different for 32 bit and 64 bit x86
 sp-$(CONFIG_X86_32) := esp
 sp-$(CONFIG_X86_64) := rsp
-- 
1.7.1